### PR TITLE
OCPBUGS-33234: azure: bump profile used for network

### PR DESCRIPTION
**Pull request by r4f4**
It seems Azure has added support for multiple address prefixes for a subnet. This means that getting subnet details might return only `addressPrefixes` while `addressPrefix` is `nil`. In such cases, the Installer will fail with a SIGSEGV:
```
$ ../openshift-install create manifests --dir .
INFO Credentials loaded from file "/root/.azure/osServicePrincipal.json"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2f7d2e9]

goroutine 1 [running]:
github.com/openshift/installer/pkg/asset/installconfig/azure.validateSubnet({0xc000e1ade8?, 0x237b46b8?}, 0xc000122018?, 0xc000abee40, {0xc000e29d40, 0x28}, {0xc000001080, 0x1, 0x4})
	/go/src/github.com/openshift/installer/pkg/asset/installconfig/azure/validation.go:529 +0x69
github.com/openshift/installer/pkg/asset/installconfig/azure.validateNetworks({0x237e1268, 0xc000015f78}, 0xc0000ced00, {0xc000001080, 0x1, 0x4}, 0xc000a4cde0)
	/go/src/github.com/openshift/installer/pkg/asset/installconfig/azure/validation.go:512 +0x2ff
[...]
```
Let's bump the profile version used from 2018-03-01 to 2020-09-01 so we can have access to both properties and select the appropriate one.

### CODE REVIEW

**Comment by r4f4** [comment]
/hold
I'll fix the unit issues after I've gotten some feedback from current job runs.

**Comment by r4f4** [comment]
I don't think we'll be able to validate `addressPrefixes` in CI, since it's possible the feature is not enabled in the subscription. At least it's not for the dev account:
```
$ az feature show --namespace Microsoft.Network -n AllowMultipleAddressPrefixesOnSubnet --subscription "OpenShift Dev Catchall"
{
  "id": "/subscriptions/<redacted>/providers/Microsoft.Features/providers/Microsoft.Network/features/AllowMultipleAddressPrefixesOnSubnet",
  "name": "Microsoft.Network/AllowMultipleAddressPrefixesOnSubnet",
  "properties": {
    "state": "NotRegistered"
  },
  "type": "Microsoft.Features/providers/features"
}
```

**Comment by r4f4** [comment]
/hold cancel

Update: fixed the unit tests, updated the vendored code to remove the old profile and added a note in the code about `addressPrefixes`.

File: pkg/asset/installconfig/azure/validation.go [hunk 1]
```diff
@@ -526,7 +526,19 @@ func validateNetworks(client API, p *aztypes.Platform, machineNetworks []types.M
 func validateSubnet(client API, fieldPath *field.Path, subnet *aznetwork.Subnet, subnetName string, networks []types.MachineNetworkEntry) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	subnetIP, _, err := net.ParseCIDR(*subnet.AddressPrefix)
+	var addressPrefix string
+	switch {
+	case subnet.AddressPrefix != nil:
```
Comment (barbacbd) [inline pkg/asset/installconfig/azure/validation.go:531]: Is there ever a case where both fields will be filled out?

Comment (r4f4) [reply pkg/asset/installconfig/azure/validation.go:531]: I don't think so, but who knows. The CU has the feature enabled, so they get `addressPrefixes` only. Our dev account doesn't have the feature, so we get `addressPrefix` only. I could not find any azure docs about this.

**Comment by r4f4** [comment]
/override ci/prow/okd-images
Known issue

**Pull request merged**